### ISSUE TITLE
macOS High Sierra 10.13 success

### ISSUE THREAD

**Issue by mkaa00x** [syscl/XPS9350-macOS#128]
I used the base of this project to boot into macOS Sierra 10.13 and so far I have about 90% of it working with my current progress displayed [here](https://www.tonymacx86.com/threads/dell-xps-13-9350-macos-high-sierra-public-beta-install.226059/)

I don't know if you are willing to work on a beta version of the OS or not but I'd still like some help on getting bluetooth, USB C (display and USB), and the preference pane for the trackpad (never shows up) to work. Is there anything I can provide you with to help?

**Comment by syscl** [comment]
Please try the latest deploy and it solved the trackpad issue.

Report if the latest VoodooPS2 works.

Thank you,
syscl

**Comment by mkaa00x** [comment]
Like on 10.13? because a lot of things are different which I'm pretty sure your deploy.sh will overwrite.

**Comment by syscl** [comment]
Just try the latest VoodooPS2 then report.

Thank you,
syscl

**Comment by mkaa00x** [comment]
Works like a charm. All functionality back in preferences pane.

**Comment by syscl** [comment]
So you mean we need a new BrcmFirmwareRepo.kext for 10.13+?

Thank you in advance,
syscl

**Comment by mkaa00x** [comment]
Nope, I somehow had an outdated one. Replacing newest BrcmFirmwareRepo.kext that was available on this git solved BT. Wifi works oob. As a side note: If you want Sound in 10.13 the best method I have found so far is to inject layout id 13 in DSDT and remove all kexts to patch for AppleHDA (since it's now AppleALC + Lilu and AppleHDA compatibility hasn't been added for 10.13)

**Comment by syscl** [comment]
@mkaa00x I am those who use AppleALC when AppleALC was once released. LayoutID has nothing to do with sound quality. I will try to update my sound driver.

syscl

**Comment by mkaa00x** [comment]
For me - the sound just didn't work at all until I followed Rehab man's instructions and set it to 13. Let me know how it goes.

**Comment by syscl** [comment]
Give me a few minutes, I manage to update the Resource for audio ...

syscl

**Comment by syscl** [comment]
@mkaa00x Please try the latest deploy.sh with layout-id = 1, and report the sound work or not.

Thank you in advance,
syscl

**Comment by syscl** [comment]
Please try this version of AppleALC, I optimized the code inside it and made it much faster.

You can use layout-id = 1 or layout-id=13.

[AppleALC.kext.zip](https://github.com/syscl/XPS9350-macOS/files/1199125/AppleALC.kext.zip)
syscl

**Comment by mkaa00x** [comment]
Hi, sorry, hadn't been on github lately. I tried it out, layout-id=13 is the only one that works for me. Either that or 28. AppleALC.kext that you provided works as flawlessly as before. I haven't tested with my headphones but it fixed a small error where my Audio devices' names would be in a foreign language (I'm assuming it's Singaporean)

**Comment by syscl** [comment]
Hi, how about layout = 1 with these AppleALC?

syscl

**Comment by flipphos** [comment]
Hi I have succeeded with ALC887 on the 10.13 public beta with the following:

1. latest Lilu and AppleALC
2. Boot Clover with -alcbeta parameter
3. good luck

I hope it works for other ALC models as well. Please try and feedback. Thanks!

Cheers,
Hong

**Comment by mkaa00x** [comment]
@syscl nope - layout-id didn't work for me.

**Comment by flipphos** [comment]
@mkaa00x 
Hi I have succeeded with ALC887 on the 10.13 public beta4 with the following:

1. Install latest Lilu and AppleALC under clover
2. Boot Clover with -alcbeta parameter
3. Good luck

I hope it works for your ALC model on the XPS 9350 as well. Please try and feedback if you are interested. Thanks!

Cheers

**Comment by mkaa00x** [comment]
I've got a working configuration but only if I set layout-id to either 13 (0x0D) or 28 (0x1C)
